examples/computer_vision/shiftvit_pytorch/README.md

```diff
@@ -0,0 +1,66 @@
+PyTorch API. Model code is imported from the ShiftViT authors' [github implementation](https://github.com/microsoft/SPACH).
+
+## Files
+
+### Python
+
+* `model_def.py`: core code containing the Determined AI `PyTorchTrial` subclass.
+* `data.py`: handles the
+[MNIST](http://pytorch.org/vision/main/generated/torchvision.datasets.MNIST.html#torchvision.datasets.MNIST),
+[CIFAR10](http://pytorch.org/vision/main/generated/torchvision.datasets.CIFAR10.html),
+and [ImageNet](https://www.image-net.org) datasets.
+* `generate_blob_list.py`: scripts to pre-process GCS blobs (increases ImageNet speed)
+
+### Bash
+* `startup-hook.sh`: startup script run by Determined within every launched container. Installs additional dependencies
+and clones the ShiftViT repo.
+
+### YAML
+Configuration files:
+* `const_mnist.yaml`: Train on MNIST using a single slot and fixed hyperparameters.
+* `distributed_mnist.yaml`: Train on MNIST using a multiple slots and fixed hyperparameters.
+* `adaptive_mnist.yaml`: MNIST hyperparameter search using ASHA.
+* `const_cifar10.yaml`: Train on CIFAR10 using a single slot and fixed hyperparameters.
+* `distributed_cifar10.yaml`: Train on CIFAR10 using a multiple slots and fixed hyperparameters.
+* `adaptive_cifar10.yaml`: CIFAR10 hyperparameter search using ASHA.
+* `distributed_imagenet.yaml`: Train on ImageNet using a multiple slots and fixed hyperparameters,
+following the ShiftViT training defaults.
+
+## Data
+The MNIST and CIFAR10 datasets are downloaded via
+[`torchvision` `Dataset` objects](https://pytorch.org/vision/stable/datasets.html).
+ImageNet is expected to have been downloaded by the user and placed in a GCS bucket, along with the
+outputs of `generate_blob_list.py`.
+
+## To Run
+If you have not yet installed Determined, installation instructions can be found
+under `docs/install-admin.html` or [here](https://docs.determined.ai/latest/index.html).
+
+A constant MNIST trial can be run via `det -m <master host:port> experiment create -f
+const_mnist.yaml .`. The other configurations can be run by specifying the appropriate
+configuration file in place of `const_mnist.yaml`.
+
+## Results
+
+### MNIST and CIFAR10
+
+MNIST and CIFAR10 trial data using GCP clusters with four T4 GPUs per node and the above `yaml`
+configs:
+
+| Config File                | Accuracy (%) |
+|----------------------------|--------------|
+| `const_mnist.yaml`         | 99.2         |
+| `distributed_mnist.yaml`   | 99.1         |
+| `const_cifar10.yaml`       | 81.7         |
+| `distributed_cifar10.yaml` | 84.4         |
+
+### ImageNet Timing
+
+The graphic below demonstrates the timing results for training the ShiftVit model on ImageNet data
+using 1, 2, 4, and 8 GPUs.  Each model was trained on a fixed number of records and the speed-up
+due to distributed training is evident. The durations are measured relative to the beginning of
+of each experiment.
+![ShiftVit Timing Tests](ShiftVitTimingTest.png)
```

Comment:
is this showing how much time it takes to train on a given number of records for 1, 2, 4, and 8 GPUs? if so, i expect the curves to be more linear than power law scaling.

Reply:
Correct, and this was something I wanted to ask you about.

The plot shows the timing from the very start of the trial, so the initial portion of the curve shows delays from initializing everything, but it looks like the curves all turn linear(-ish) by the end.  These tests don't really run for all that long, so I was wondering if you're seeing like a caching-effect in the middle of the curve.  I could run the tests for longer, if you think it's worth it?